### PR TITLE
[stable/gangway] - add option to extend env

### PR DESCRIPTION
**Pull request by PixelJonas**
#### Is this a new chart
no

#### What this PR does / why we need it:
currently, there is no option to add custom environment variables to the deployment config of gangway. Since a few of the configuration options are sensitive (`GANGWAY_SESSION_SECURITY_KEY`, `GANGWAY_CLIENT_SECRET`) I think we should be able to set these via pre-existing secrets.

#### Which issue this PR fixes
none

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/gangway/README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ At a minimum you *must* configure any of the values marked as **required** in th
...
 | -------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------- |
 | `affinity`                       | List of affinities (requires Kubernetes >=1.6)                                                                                                                                                                  | `{}`                                               |
+| `env`                   | Optional list of environment variables                                                                                                                                                                                           | `[]`                                               |
```
Comment (rk295) [inline stable/gangway/README.md]: Might want to add a note explaining that `GANGWAY_SESSION_SECURITY_KEY` is already defined and you cannot override.

File: stable/gangway/templates/deployment.yaml [hunk 2]
```diff
@@ -35,6 +35,9 @@ spec:
         - -config
         - /gangway/gangway.yaml
         env:
+{{ if .Values.env -}}
```
Comment (rk295) [inline stable/gangway/templates/deployment.yaml]: Changing this to:
```suggestion
{{- if .Values.env }}
```

Means the `yaml` doesn't have a blank line when `env` isn't set. Trivial I'll admit but slightly nicer reading for anybody previewing the `yaml`!

**Review by rk295 — APPROVED** [review]
Two trivial comments, otherwise fine.

File: stable/gangway/README.md (hunk 1 above)
Comment (PixelJonas) [reply stable/gangway/README.md]: we could also add the `GANGWAY_SESSION_SECURITY_KEY` as default to the `values.yaml` so it stays consistent?

Comment (rk295) [reply stable/gangway/README.md]: I like the idea, but I think the problem will be this bit:

```
              name: {{ include "gangway.fullname" . }}-key
```

You can't include template fns. in `values.yaml`. I'd leave it and add a warning, nobody has ever asked for that to be made an option so I'm tempted to say lets cross that bridge when we come to it!

Comment (PixelJonas) [reply stable/gangway/README.md]: yeah you are totally right - I will update the PR today

Comment (PixelJonas) [reply stable/gangway/README.md]: srry, for the delay - this issue is still active for me and I'll try to make the change as soon as possible

**Comment by PixelJonas** [comment]
since https://github.com/helm/charts/pull/20486 is already merged there is now a way to specify environment variables. introducing this change would be a breaking change now because my solutions do not map `key: value` to environment variables but uses the `toYaml`.

the existing PR is no solution to my problem though since I need to add an environment variable passed on a secret. I'll try to fork again from the new master and maybe adds a `secretEnv` option wich also maps secrets to environment variables.

**Pull request closed without merging**